test/e2e/framework/service_util.go

```diff
@@ -1550,7 +1550,7 @@ func checkAffinityFailed(tracker affinityTracker, err string) {
 // return false only in case of unexpected errors.
 func CheckAffinity(jig *ServiceTestJig, execPod *v1.Pod, targetIP string, targetPort int, shouldHold bool) bool {
 	targetIPPort := net.JoinHostPort(targetIP, strconv.Itoa(targetPort))
-	cmd := fmt.Sprintf(`wget -qO- http://%s/ -T 2`, targetIPPort)
+	cmd := fmt.Sprintf(`curl -s http://%s/ -m 3`, targetIPPort)
```

Comment:
LGTM :+1: 
But why do we need to change the timeout form 2 seconds to 3 seconds ?
What's more, `wget` will retry 20 times by default. And This makes the code more robust.
Part of the output of `man wget`:
```
       -t number
       --tries=number
           Set number of tries to number. Specify 0 or inf for infinite retrying.  The default is to retry 20 times, with the exception of fatal errors like "connection refused" or "not found" (404), which are not
           retried.
```
So I suggest adding this parameter for `curl`:
```suggestion
	cmd := fmt.Sprintf(`curl --retry 20 -s http://%s/ -m 2`, targetIPPort)
```

Reply:
>So I suggest adding this parameter for curl:

_When curl is about to retry a transfer, it will first wait one second and then for all forthcoming retries it will double the waiting time until it reaches 10 minutes which then  will be the delay between the rest of the retries._

`--retry 5` might be sufficient with this exponential backoff or `--retry 20 --retry-delay 2` would stand good here. Please make your views @SataQiu .

Reply:
Thank you for your explanation @mgdevstack 
I think both are fine. We just need to increase the robustness of our code appropriately.
@neolit123 What do you think about this?